python/python/lance/dataset.py

```diff
@@ -3956,6 +3957,9 @@ def write_dataset(
         These row ids are stable after compaction operations, but not after updates.
         This makes compaction more efficient, since with stable row ids no
         secondary indices need to be updated to point to new row ids.
+    auto_cleanup_options: optional, dict
+        Config options for automatic cleanup of the dataset.
+        Note: this config options only take effect when creating a new dataset.
```

Comment:
Users won't know how to use this.  I think we will need to expand on this comment quite a bit to explain what auto cleanup is and how it works.  Also, we should be clear that, if this is set to None, then there will be no automatic cleanup.